report-viewer/src/views/ComparisonView.vue

```diff
@@ -0,0 +1,225 @@
+<template>
+  <div class="container">
+    <button id="show-button" :class="{hidden : !hideLeftPanel}" @click="togglePanel" title="Show sidebar">
+      <img src="@/assets/double_arrow_black_24dp.svg" alt="show">
+    </button>
+    <div id="sidebar" :class="{ hidden : hideLeftPanel }">
+      <div class="title-section">
+        <h1>JPlag Comparison</h1>
+        <button id="hide-button" @click="togglePanel" title="Hide sidebar">
+          <img src="@/assets/keyboard_double_arrow_left_black_24dp.svg" alt="hide"></button>
+      </div>
+      <TextInformation label="Submission 1" :value="id1" :anonymous="store.state.anonymous.has(id1)"/>
+      <TextInformation label="Submission 2" :value="id2" :anonymous="store.state.anonymous.has(id2)"/>
+      <TextInformation label="Match %" :value="comparison.match_percentage"/>
+      <MatchTable :matches="comparison.allMatches" :id1="id1" :id2="id2" @match-selected="showMatch" />
+    </div>
+    <FilesContainer :matches="comparison.matchesInFirstSubmission" :files="filesOfFirst" :container-id="1"
+                    files-owner="Submission 1"
+                    @toggle-collapse="toggleCollapseFirst"
+                    @line-selected="showMatchInSecond"/>
+    <FilesContainer :matches="comparison.matchesInSecondSubmissions" :files="filesOfSecond" :container-id="2"
+                    files-owner="Submission 2"
+                    @toggle-collapse="toggleCollapseSecond"
+                    @line-selected="showMatchInFirst"/>
+  </div>
+</template>
+
+<script>
+import { defineComponent, ref } from "vue";
+import {generateLineCodeLink} from "@/utils/Utils";
+import store from "@/store/store";
+import router from "@/router";
+import TextInformation from "@/components/TextInformation";
+import MatchTable from "@/components/MatchTable";
+import {ComparisonFactory} from "@/model/factories/ComparisonFactory";
+import FilesContainer from "@/components/FilesContainer";
+
+export default defineComponent({
+  name: "ComparisonView",
+  components: {FilesContainer, MatchTable, TextInformation },
+  props: {
+    id1: {
+      type: String,
+    },
+    id2: {
+      type: String,
+    },
+  },
+  setup(props) {
+    /**
+     * Name of the comparison file. Comparison files should be named {ID1}-{ID2}
+     * @type {string}
+     */
+    const fileName1 = props.id1.concat("-").concat(props.id2)
+    const fileName2 = props.id2.concat("-").concat(props.id1)
+
+    let comparison;
+    //getting the comparison file based on the used mode (zip, local, single)
+    if(store.state.local) {
+      try {
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        comparison = ComparisonFactory.getComparison(require(`../files/${fileName1}.json`))
```

Comment:
Again, good question whether `require` is the correct way to load json files at runtime...

Reply:
Well I couldn't find any alternative way except for _fs_, for which I have read that is server-side and that is why I avoided it.

Reply:
I wonder whether this would be actually a problem here. Not sending files to server is only a real problem when dealing with files that are drag 'n' dropped by the user client-side (i.e., zip files). Files that are read from the system are already on the servers side. But maybe I'm wrong here, please correct me.